Remotion/Data/DomainObjects/Mapping/ClassDefinition.cs

```diff
@@ -446,61 +223,58 @@ private void FillAllDerivedClasses (List<ClassDefinition> allDerivedClasses)
       }
     }
 
-    private void CheckPropertyDefinitions (IEnumerable<PropertyDefinition> propertyDefinitions)
+    public override string ToString ()
+    {
+      return $"{GetType().GetFullNameChecked()}: {ID}";
+    }
+
+    protected override void CheckPropertyDefinitions (IEnumerable<PropertyDefinition> propertyDefinitions)
     {
       foreach (var propertyDefinition in propertyDefinitions)
       {
-        if (!ReferenceEquals(propertyDefinition.ClassDefinition, this))
+        if (!ReferenceEquals(propertyDefinition.TypeDefinition, this))
         {
-          throw CreateMappingException(
-              "Property '{0}' cannot be added to class '{1}', because it was initialized for class '{2}'.",
-              propertyDefinition.PropertyName,
-              _id,
-              propertyDefinition.ClassDefinition.ID);
+          throw new MappingException(
```

Comment:
why did you change this away from the factory method? Is this just because you want to use the interpolated strings? in that case, interpolated strings should wait until we have interpolated overloads and can keep the factory methods.

Reply:
What exactly is the benefit of having a factory method in this case? It just hides the type of exception that from the reader. If the method provided any conditional logic or values that must not provided I would argue for the factory method. But as it stands I don't see the value for it.

Reply:
It's a change that's completely unrelated to the current task. There's several factory methods for MappingException (and possibly some other exception types in Data.DomainObjects) and the descision to remove them should be consistent accross the (related) code base. Originally, the reason was likely to remove the string.Format calls from the call sites and perhaps also to remove the code duplication of choosing the exception type. Those descisions go back to 2004 so not everything's still valid in today's world. Either way, though, the change should be done separately and consistently on the code base, unless there's a benefit to the current task.

Reply:
I understand the need for the factory methods if you have complex stuff that you do not want to repeat everywhere. In this case the inlined code is probably shorter than the current method invocation while also making it clear what is happening without needing to jump to the helper method.

I don't quite get the argument of we should do it everywhere or nowhere at all. I am not for removing the helper methods just for the fun but in this case the helper is just not necessary and, in my opinion, makes the code less readable. Since all the helper methods are redeclared on each class why do we need to change all at once?

Reply:
My reasoning is that there is no need to change this now since the code doesn't depend on this - we don't even have to change the API etc. To me, this makes it into a completely unrelated cleanup we can do any time, but we should do it in a dedicated commit (and task). And when we do that, we can cover the other 5 or so CreateMappingExceptions, too, since it's just 5 minutes more work. But, by doing it in a separate task, we isolate the change and make it easily tracable.